### PR TITLE
Update timestamping tutorial according to 0.10

### PR DESCRIPTION
**Pull request by ivan-ochc**

### CODE REVIEW

File: src/get-started/timestamping-tutorial.md [hunk 1]
```diff
@@ -42,24 +42,33 @@ publish = false
 authors = ["Your name <your@email.com>"]
```
Comment (aleksuss) [inline src/get-started/timestamping-tutorial.md]:
```suggestion
edition = "2018"
```

File: src/get-started/timestamping-tutorial.md [hunk 2]
```diff
@@ -125,6 +132,78 @@ const TIMESTAMPING_SERVICE: u16 = 130;
...
+
+Create new `proto` directory inside `src`:
+```bash
+mkdir proto
+```
+
+Inside `proto` directory create two files: `timestamping.proto` and `mod.rs`.
+Define protobuf structures in `timestamping.proto`:
+
+```protobuf
```
Comment (aleksuss) [inline src/get-started/timestamping-tutorial.md]:
```suggestion
```proto
```

File: src/get-started/timestamping-tutorial.md [hunk 3]
```diff
@@ -70,35 +79,32 @@ where we indicate the external libraries required for our service.
...
     extern crate exonum_time;
     #[macro_use]
+    extern crate exonum_derive;
+    #[macro_use]
     extern crate failure;
     #[macro_use]
     extern crate log;
-    extern crate serde;
     #[macro_use]
     extern crate serde_derive;
```
Comment (aleksuss) [inline src/get-started/timestamping-tutorial.md]: If we won't change rust edition in `Cargo.toml` to 2015 then `extern crate ...` is redundant.

File: src/get-started/timestamping-tutorial.md [hunk 4]
```diff
@@ -125,6 +132,78 @@ const TIMESTAMPING_SERVICE: u16 = 130;
...
+pub use self::timestamping::{Timestamp, TimestampEntry, TxTimestamp};
+
+include!(concat!(env!("OUT_DIR"), "/protobuf_mod.rs"));
+
+use exonum::proto::schema::*;
+```
+
+## Generate rust structs from proto files
+
+In the project root create `build.rs` file. Add the following code to `build.rs`:
```
Comment (aleksuss) [inline src/get-started/timestamping-tutorial.md]:
```suggestion
In the project root directory create `build.rs` file. Add the following code to `build.rs` file:
```

File: src/get-started/timestamping-tutorial.md [hunk 5]
```diff
@@ -125,6 +132,82 @@ const TIMESTAMPING_SERVICE: u16 = 130;
...
+    protobuf_generate(
+        "src/proto",
+        &["src/proto", &exonum_protos],
+        "protobuf_mod.rs",
+    );
+}
+```
+
+After successful run output directory will contain *.rs for each
+*.proto file in `"src/proto/**/"` and `example_mod.rs` which will
```
Comment (dvush) [inline src/get-started/timestamping-tutorial.md]: Why both ` and " in `"src/proto/**/"`?

File: src/get-started/timestamping-tutorial.md [hunk 6]
```diff
@@ -164,24 +267,37 @@ The `Timestamp` structure includes the hash of the submitted file and its
...
+    pub time: DateTime<Utc>,
+}
 
-        /// Timestamp time.
-        time: DateTime<Utc>,
+impl TimestampEntry {
+    /// New TimestampEntry.
+    pub fn new(timestamp: Timestamp,
+               &tx_hash: &Hash,
+               time: DateTime<Utc>) -> Self {
```
Comment (dvush) [inline src/get-started/timestamping-tutorial.md]: Bad code format.
Functions with a lot of args formatted like this:
```rust
    pub fn sign(
        pk: &PublicKey,
        &to: &PublicKey,
        amount: u64,
        seed: u64,
        sk: &SecretKey,
    ) -> Signed<RawTransaction> {
        Message::sign_transaction(
            Self { to, amount, seed },
            CRYPTOCURRENCY_SERVICE_ID,
            *pk,
            sk,
        )
    }
```

File: src/get-started/timestamping-tutorial.md [hunk 7]
```diff
@@ -398,26 +523,22 @@ signature, while `execute` performs the following operations:
...
 impl Transaction for TxTimestamp {
-    fn verify(&self) -> bool {
-        self.verify_signature(self.pub_key())
-    }
-
-    fn execute(&self, fork: &mut Fork) -> ExecutionResult {
-        let time = TimeSchema::new(&fork)
+    fn execute(&self, mut context: TransactionContext) -> ExecutionResult {
+        let tx_hash = context.tx_hash();
+        let time = TimeSchema::new(&context.fork())
```
Comment (dvush) [inline src/get-started/timestamping-tutorial.md]: `TimeSchema` or `Schema`? Names are not consistent.

File: src/get-started/timestamping-tutorial.md [hunk 8]
```diff
@@ -164,24 +267,37 @@ The `Timestamp` structure includes the hash of the submitted file and its
...
-    /// Timestamp entry.
-    struct TimestampEntry {
-        /// Timestamp data.
-        timestamp: Timestamp,
-
-        /// Hash of transaction.
-        tx_hash: &Hash,
+/// Timestamp entry.
+#[derive(Clone, Debug, ProtobufConvert)]
+#[exonum(pb = "proto::TimestampEntry", serde_pb_convert)]
```
Comment (dvush) [inline src/get-started/timestamping-tutorial.md]: Maybe some notes about `serde_pb_convert`. See PR #289

File: src/get-started/timestamping-tutorial.md [hunk 9]
```diff
@@ -297,43 +413,54 @@ in [src/schema.rs][src/schema.rs] and the `TIMESTAMPING_SERVICE` constant
...
+}
 
-            /// Timestamp content.
-            content: Timestamp,
-        }
+impl TxTimestamp {
+    #[doc(hidden)]
+    pub fn sign(author: &PublicKey,
+                content: Timestamp,
+                key: &SecretKey) -> Signed<RawTransaction> {
```
Comment (dvush) [inline src/get-started/timestamping-tutorial.md]: Wrong format.

File: src/get-started/timestamping-tutorial.md [hunk 10]
```diff
@@ -125,6 +132,82 @@ const TIMESTAMPING_SERVICE: u16 = 130;
 pub const SERVICE_NAME: &str = "timestamping";
 ```
 
+## Define protobuf structures
+
+Create new `proto` directory inside `src`:
```
Comment (aleksuss) [inline src/get-started/timestamping-tutorial.md]: a new ?

File: src/get-started/timestamping-tutorial.md (hunk 7 above)
Comment (aleksuss) [reply src/get-started/timestamping-tutorial.md]: @dvush `TimeSchema` is a type from time service. It's ok I guess.

File: src/get-started/timestamping-tutorial.md [hunk 11]
```diff
@@ -40,26 +40,35 @@ name = "exonum-demo-timestamping"
 version = "0.0.0"
```
Comment (aleksuss) [inline src/get-started/timestamping-tutorial.md]: AFAIK a `cargo new` command generates "0.1.0" version by default.

File: src/get-started/timestamping-tutorial.md [hunk 12]
```diff
@@ -70,35 +79,32 @@ where we indicate the external libraries required for our service.
 ??? note "Imports"
 
     ```rust
-    extern crate chrono;
-    #[macro_use]
     extern crate exonum;
```
Comment (aleksuss) [inline src/get-started/timestamping-tutorial.md]: I propose to get rid of the `extern crate `s according Rust 2018

**Pull request closed without merging**